charts/redpanda/render.go

```diff
@@ -0,0 +1,97 @@
+// Copyright 2025 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+// +gotohelm:filename=_render.go.tpl
+package redpanda
+
+import (
+	"fmt"
+
+	"github.com/cockroachdb/errors"
+	appsv1 "k8s.io/api/apps/v1"
+
+	"github.com/redpanda-data/redpanda-operator/charts/console/v3"
+	"github.com/redpanda-data/redpanda-operator/gotohelm/helmette"
+	redpandav1alpha3 "github.com/redpanda-data/redpanda-operator/operator/api/redpanda/v1alpha3"
+	"github.com/redpanda-data/redpanda-operator/pkg/kube"
+)
+
+// This serves as a running list of places in the v2 chart rendering functions that will need an update to be NodePool-aware:
+// - configmap.tpl.go: RedpandaConfigFile (node configuration), RPKProfile (awareness of other brokers)
+// - console.tpl.go: ConsoleConfig (uses ordinals based on replicas)
+// - poddisruptionbudget.go: PodDisruptionBudget (calculated based off of single statefulset)
+// - service.loadbalancer.go: LoadBalancerServices (uses ordinals based on replicas)
+// - ? service.nodeport.go: NodePortService (uses label selector for one statefulset)
+// - ? service_internal.go: ServiceInternal (uses label selector for one statefulset)
+// - secrets.go: SecretSTSLifecycle (uses replicas for preStopHook bypass), SecretFSValidator (uses StatefulSet.InitContainers and replicas), SecretConfigurator (uses replicas)
+// - statefulset.go: no explanation needed
+
+// +gotohelm:ignore=true
```

Comment:
I'm fairly certain that this annotation can be placed at the package level and then gotohelm will skip the file entirely.

You could also use a build tag, IIRC there's a gotohelm buildtag.